### DISCUSSION TITLE
Support M503 plugin

### DISCUSSION THREAD

**Discussion by doucettom** [OctoFarm/OctoFarm#1112] [Feature Request]
### OctoFarm Version

1.2

### OctoPrint Version(s)

1.7.3

### pm2 or docker?

pm2

### Operating System

Ubuntu

### Location of the problem

Not sure

### Description of the problem

```bash
...
It would be nice if we could visualize/backup/restore the values from M503 which contains esteps, pids etc.
```

### Steps taken to reproduce

```bash
1.
2.
3.
...
```

### What did you expect to happen

```bash
...
```

### To avoid the problem I had to

```bash
...
```

**Comment by NotExpectedYet** [comment]
Is the plugin a ok way forward for this? Seems most of the heavy lifting parsing the gcode output is handled by that. I'm not entirely sure how well that will go over the API. 

I was thinking it could work as follows: 

 - [ ] Add detection trigger for EPROM editor plugin Turns on another automation setting
 - [ ] Add setting whether to run a M503 on connected printers periodically (requires EPROM editor plugin)
 - [ ] Add in to OP dropdown to install ALL OF supported plugins (except Filament Manager)
 - [ ] Clean up settings page
   - [ ] Users get a "Settings" button leading to Client page. 
   - [ ] Administration page, pull out tabs into list instead. 
   - [ ] Timeout -> Printer Settings
   - [ ] Background, Monitoring views into Client page visible by admin only. 
   - [ ] System -> Server tab only. Add in .env editing with restart required flag to enable. 


Future Ideas (out of scope): 
- [ ] Some form of EPROM editor? dunno.
